src/cpp/session/modules/SessionReticulate.R

```diff
@@ -79,6 +79,45 @@
    return(TRUE)
 })
 
+.rs.addFunction("reticulate.matplotlib.showHook", function(...)
+{
+   # read device size
+   size <- dev.size(units = "in")
+   width <- size[1]; height <- size[2]
+   
+   # TODO: handle high-DPI displays
+   dpi <- 72L
+   
+   # TODO: get device requested from matplotlib?
+   # TODO: handle HTML content?
+   path <- tempfile("matplotlib-plot-", fileext = ".png")
+   
+   plt <- reticulate::import("matplotlib.pyplot", convert = TRUE)
+   
+   # resize the figure
+   figure <- plt$gcf()
+   figure$set_dpi(dpi)
+   figure$set_size_inches(width, height)
+   plt$savefig(path, dpi = figure$dpi)
+   
+   # now, read that in and create an R plot using e.g. image
+   data <- png::readPNG(path, native = TRUE, info = TRUE)
```

Comment:
Do we need to prompt for installation of the png package here? (or should we just make that a dependency of reticulate?)

Reply:
We can add reticulate as one of the packages in the list of dependencies for reticulate within the DependencyManager (so it all comes in one shot).